corehq/apps/hqcase/api/get_list.py

```diff
@@ -93,13 +93,14 @@ def get_list(domain, params):
             raise UserError(f"'{key}' is not a valid parameter.")
 
     es_result = query.run()
+    hits = es_result.hits
     ret = {
         "total": es_result.total,
-        "cases": [serialize_es_case(case) for case in es_result.hits],
+        "cases": [serialize_es_case(case) for case in hits],
     }
-    cases_in_result = len(es_result.hits)
+    cases_in_result = len(hits)
     if cases_in_result and es_result.total > cases_in_result:
```

Comment:
Does this mean that sometimes the `next` link will return an empty result set? Do you also need to check `cases_in_result == page_size` (i.e. that this isn't the last page of results)?
Maybe I've misunderstood how that bit happens here.

Reply:
I don't think so, do you mean if the total results is a multiple of the page size?  I think regardless of that this shouldn't give a `next` link with an empty set.  Here are two examples:

## Scenario 1:
Request | total | page_size | cases_in_result
--|---|---|---
1 | 40 | 20 | 20
2 | 20 | 20 | 20

## Scenario 2:
Request | total | page_size | cases_in_result
--|---|---|---
1 | 48 | 20 | 20
2 | 28 | 20 | 20
3 | 8 | 20 | 8

In both cases, `total == cases_in_result` for the last page.  This could also use `page_size` in place of `cases_in_result` and give the same behavior.  Am I overlooking an edge case?

Reply:
Aha! I missed the fact that `cases_in_result` was the total number of cases for that particular "page", not the total for the whole search.

Reply:
Ah yep - we don't have that information unfortunately, since cases older than the start position don't match the ES filter.  We'd have to keep track of the "original" query for that, I think.

Reply:
Yeah, I do think this makes the `total` field a bit misleading though, since it's actually "total left to be fetched if you follow all the links".

Reply:
I was thinking the same but after thinking about it a bit more I thought that it does make sense since each page is updating the `indexed_on` filter. But it took me a minute or two to get there so maybe that's an indication that it is confusing. 

I think that was one of the reasons people use encoded cursors because they can add additional information into them like the total number of records.

Reply:
Just making a point that this should be well documented. If you weren't really paying attention to what the `next` link does, it might be hard to reason about why the total keeps changing.